app/templating/summary/group.py

```diff
@@ -1,29 +1,57 @@
+from collections import defaultdict
+
 from app.templating.summary.block import Block
+from app.helpers.schema_helpers import get_group_instance_id
+from app.questionnaire.location import Location
+from app.templating.template_renderer import renderer
 
 
 class Group:
 
-    def __init__(self, group_schema, path, answer_store, metadata, schema):
-        self.id = group_schema['id']
-        self.title = group_schema.get('title')
-        self.blocks = self._build_blocks(group_schema, path, answer_store, metadata, schema)
+    def __init__(self, group_schema, path, answer_store, metadata, schema, group_instance, schema_context):
+        self.id = group_schema['id'] + '-' + str(group_instance)
+        location = Location(group_schema['id'], group_instance, group_schema['blocks'][0]['id'])
+        self.schema_context = schema_context
+        self.schema_context['group_instance_id'] = get_group_instance_id(schema, answer_store, location)
```

Comment:
As discussed, I've removed the modification of schema_context here since it should refer to current location.

Since `schema_context` is passed into render, that group_instance_id will also represent the current location. It's not currently used though.

Reply:
@ajmaddaford I could have sworn I searched all the templates for 'group_instance_id' and didn't find any, but apparently not. My bad.

`group_instance_id` is used in the test_repeat_until_summary schema in a number of places. The most problematic of these is in the question titles. Of course currently, these work properly on the question itself, but don't display on the summary. 

The easy fix is to override group_instance_id in the call to render. Thoughts?

Reply:
I thought I'd searched the templates before I made the comment as well! I agree, for now we should override it next to the render call. I'll raise a card for reviewing how we use the schema context.